feature/gribi/ate_tests/base_hierarchical_route_installation_test/base_hierarchical_route_installation_test.go

```diff
@@ -102,68 +113,118 @@ func configInterfaceDUT(i *oc.Interface, a *attrs.Attributes) *oc.Interface {
 	return i
 }
 
+// configureNetworkInstance creates nonDefaultVRF
+func configureNetworkInstance(t *testing.T, dut *ondatra.DUTDevice) {
+	d := &oc.Root{}
+	ni := d.GetOrCreateNetworkInstance(nonDefaultVRF)
+	ni.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
+	gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(nonDefaultVRF).Config(), ni)
+
+	if *deviations.ExplicitGRIBIUnderNetworkInstance {
+		fptest.EnableGRIBIUnderNetworkInstance(t, dut, nonDefaultVRF)
+		fptest.EnableGRIBIUnderNetworkInstance(t, dut, *deviations.DefaultNetworkInstance)
+	}
+
+	dutConfNIPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance)
+	gnmi.Replace(t, dut, dutConfNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+
+	// configure PBF in DEFAULT vrf
+	defNIPath := gnmi.OC().NetworkInstance(*deviations.DefaultNetworkInstance)
+	gnmi.Replace(t, dut, defNIPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	gnmi.Replace(t, dut, defNIPath.PolicyForwarding().Config(), configurePBF())
+}
+
 // configureDUT configures port1 and port2 on the DUT.
 func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
 	d := gnmi.OC()
+	for p, dp := range dutPorts {
+		p1 := dut.Port(t, p)
+		i1 := &oc.Interface{Name: ygot.String(p1.Name())}
+		gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), configInterfaceDUT(i1, &dp))
+		if *deviations.ExplicitPortSpeed {
+			fptest.SetPortSpeed(t, p1)
+		}
+		if *deviations.ExplicitIPv6EnableForGRIBI {
```

Comment:
I think deviation calls should be updated across the tc to use the functions, 
if deviations.ExplicitIPv6EnableForGRIBI(dut)

Reply:
Besides this one, any other concerns from your team? @manan-patel 

Note, this deviation format update can come from another follow up PR for small incremental changes.

Reply:
I did update all deviations as manan mentioned.  Only pending dependency is in bumping ondatra version to 0.1.15 .

Reply:
testRecursiveIPv4EntrywithMACNexthop tc is failing for us at NH programming,

args.client.AddNH(t, nhIndex2, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: p.Name(), SubInterface: 0, Mac: nhMAC})

gribi.go
case "MACwithInterface":
		for _, opt := range opts {
			nh = nh.WithSubinterfaceRef(opt.Interface, opt.SubInterface).
				WithMacAddress(opt.Mac)
		}

Do we need to pass the subinterface reference here? Can we reference it only over the interface?

example:
args.client.AddNH(t, nhIndex2, "MACwithInterface", deviations.DefaultNetworkInstance(args.dut), fluent.InstalledInFIB, &gribi.NHOptions{Interface: p.Name(), Mac: nhMAC})

gribi.go
case "MACwithInterface":
		for _, opt := range opts {
			nh = nh.WithInterfaceRef(opt.Interface).
				WithMacAddress(opt.Mac)
		}

Reply:
@manan-patel , Is the issue with only sub interface "0" ?  if the interface has only subinterface "0" configured then  passing interfaceref might work but we also need to support non-zero subinterface so that was the reason I have added subinterface thinking it will work for both cases. @xw-g  any inputs??

Reply:
@manan-patel , i have added check code to send only interface when sub interface value is 0 in addNH  Could you please confirm if this works for you.